src/server/index.ts

```diff
@@ -9,28 +9,56 @@ import bodyParser from 'body-parser';
 import { configureStore } from '../shared/store';
 import serverRender from './render';
 import paths from '../../config/paths';
+import { ErrorRequestHandler, Handler, RequestHandler, Request } from 'express-serve-static-core';
+import { Store } from 'redux';
 
 require('dotenv').config();
 
 const app = express();
 
+interface RequestWithStore extends Request {
+    store?: Store;
+}
+
+const errorHandler: ErrorRequestHandler = (err, _, res) =>
+    res.status(404).json({
+        status: 'error',
+        message: err.message,
+        stack:
+            // print a nicer stack trace by splitting line breaks and making them array items
+            process.env.NODE_ENV === 'development' &&
+            (err.stack || '')
+                .split('\n')
+                .map((line: string) => line.trim())
+                .map((line: string) => line.split(path.sep).join('/'))
+                .map((line: string) =>
+                    line.replace(
+                        process
+                            .cwd()
+                            .split(path.sep)
+                            .join('/'),
+                        '.'
+                    )
+                ),
+    });
+
 // Use Nginx or Apache to serve static assets in production or remove the if() around the following
 // lines to use the express.static middleware to serve assets for production (not recommended!)
 if (process.env.NODE_ENV === 'development') {
     app.use(paths.publicPath, express.static(path.join(paths.clientBuild, paths.publicPath)));
-    app.use('/favicon.ico', (req: express.Request, res: express.Response) => {
-        res.send('');
-    });
+    app.use('/favicon.ico', (_, res) => res.send(''));
 }
 
 app.use(cors());
 
 app.use(bodyParser.json());
 
-app.use((req: any, res: express.Response, next: express.NextFunction) => {
+const addStore: RequestHandler = (req: RequestWithStore, res, next) => {
     req.store = configureStore();
```

Comment:
You impure slut!

Reply:
I think you should rather use `res.locals` to store this kind of data. http://expressjs.com/en/api.html#res.locals

Reply:
Why is this impure?

Reply:
You write to the argument of a function. That is always impure because calling your function changes the state of the inputs.

Reply:
I guess writing to `res.locals.store` instead is impure as well? If so, how do I "transport" the created store from one middleware to another? I think that's impossible here by definition to be pure because I explicitly _want to_ modify the `res.locals` object so I can use it in subsequent middleware functions?

Reply:
The configureStore() function itself is pure so I guess it shouldn't be an issue here. Is the function impure only because I modify an input parameter when the modification that takes place is pure? The modified res.locals object is the same on every request.

Reply:
You will not get this pure. It is express after all. But if you have to store your sh** somewhere, do it in `res.locals.yourShit` :)